go/chat/uithreadloader.go

```diff
@@ -563,15 +573,24 @@ func (t *UIThreadLoader) shouldIgnoreError(err error) bool {
 	return false
 }
 
-func (t *UIThreadLoader) singleFlightConv(ctx context.Context, convID chat1.ConversationID) (context.Context, context.CancelFunc) {
+func (t *UIThreadLoader) singleFlightConv(ctx context.Context, convID chat1.ConversationID,
+	reason chat1.GetThreadReason) (context.Context, context.CancelFunc) {
 	t.activeConvLoadsMu.Lock()
 	defer t.activeConvLoadsMu.Unlock()
 	convIDStr := convID.ConvIDStr()
-	if cancel, ok := t.activeConvLoads[convIDStr]; ok {
-		cancel()
+	if convLoad, ok := t.activeConvLoads[convIDStr]; ok {
+		if convLoad.reason != chat1.GetThreadReason_PUSH {
```

Comment:
why bother storing this conv load if we don't want to cancel it?

Reply:
also should this map get cleaned up after the call?

Reply:
Sure I guess that could be better. I don't think the cleanup is necessary though